test/merge.js

```diff
@@ -106,6 +106,61 @@ test('should add nested object in target', function(t) {
     t.end()
 })
 
+test('should clone source and target', function(t) {
+    var src = {
+        "b": {
+            "c": "foo"
+        }
+    }
+
+    var target = {
+        "a": {
+            "d": "bar"
+        }
+    }
+
+    var merged = merge(target, src,{clone:true})
+     
+    merged.b.c = "foo-modified"
+    merged.a.d = "bar-modifies"
+    t.equal(src.b.c, "foo") 
+    t.equal(target.a.d, "bar") 
+
```

Comment:
These two tests should probably assert that `merged` deep equals `expected`, since the two "modified" values wouldn't conflict with each other.